### PR TITLE
Upstream TorchSim model interface

### PR DESCRIPTION
**Pull request by CompRhys**
Hello, we think ts is getting to a fairly mature point now and we would like to encourge what we have called external model posture similar to ase (https://github.com/TorchSim/torch-sim/discussions/120) I have made this PR as a basic demo of what an external implementation might look like for SevenNet https://github.com/TorchSim/torch-sim/issues/429.

I made some basic changes to the CI to add linting and using uv for test env such that ts will resolve properly for > py312 whilst still pointing to the external model PR. Draft until that PR get's merged/0.5.2 release.

### CODE REVIEW

**Comment by CompRhys** [comment]
> 1. Compared to a ASE calculator interface, the interface required by TorchSim is likely to be more complex given its focus on optimization. Is there any developer documentation or guidance (or in preparation)? In my opinion, developers may need official benchmark script that compares ASE calculator and torchsim to test validity of their implementation.

This is a complete PR for the whole interface and it already includes a test factory to check the ASE calculator against the model interface for a suite of `SimState`. Here due to a divide by zero issue I omitted testing on the molecule state with non pbc boundary conditions. There were two known issues with sevennet which were that trying to cast to float64 failed and this divide by zero error when there is no cell.

I am of the opinion that the ts interface is just as simple as ASE. You take in a SimState and then return it with energy, force and stress predictions. Whether or not the model prediction is batched by an implementation isn't enforced by the interface. All the MD/Relax etc are then batched but the main speed-ups come from avoiding the serial model calls as you have with ASE.

> 2. Regarding
>    https://github.com/TorchSim/torch-sim/blob/main/torch_sim/models/interface.py
>    is this interface considered complete at this point, or is it still expected to change before the next release?

The interface as it is has been stable since July 2025 when we renamed `batch` -> `system_idx`.

> 3. Is there a planned timeline for deprecating the existing SevenNet code inside TorchSim? I need this information to add a torch sim page in our docs.

Plan is just to wrap upstream implementations to keep existing scripts backward compatible. If we did decide to remove the models from the ts namespace in the future we would likely post a similar discussion issue 3-6 months in advance.

File: sevenn/torchsim.py [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+        n_atoms_per_system = sim_state.system_idx.bincount()
+        stride = torch.cat(
+            (
+                torch.tensor([0], device=self._device, dtype=torch.long),
+                n_atoms_per_system.cumsum(0),
+            ),
+        )
+
+        data_list = []
+        for sys_idx in range(n_systems):
```
Comment (CompRhys) [inline sevenn/torchsim.py]: This implementation could be sped up using a batched neighbour list e.g. https://github.com/venkatkapil24/batch_nl or https://github.com/TorchSim/torch-sim/blob/main/torch_sim/neighbors/alchemiops.py

**Comment by CompRhys** [comment]
Hello, we just cut the 0.5.2 release which introduces the testing utils I employed here. As such this is now ready to review/merge

**Comment by CompRhys** [comment]
The test issue sees to be to do with the torch version given that I extended it to test both the oldest supported 3.8 and newest stable 3.13. Had hoped 2.4 pin would fix for 3.13 but seems that they didn't make a wheel for 3.13 with that pinned version.
